### PR TITLE
python3Packages.galois: 0.0.32 -> 0.1.1

### PR DESCRIPTION
**Pull request by ChrisPattison**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/galois/default.nix [hunk 1]
```diff
@@ -11,18 +12,24 @@
...
     owner = "mhostetter";
     repo = "galois";
     rev = "refs/tags/v${version}";
-    sha256 = "sha256-+cxRLrfqk3N9pWKCVsTxruZwMYZ5dQyKJRnrb8y+ECM=";
+    sha256 = "sha256-ekbebwsdGkINdOFI+LLdwTRBN6d8mlNmftkj5SNRKNI=";
   };
 
+  SETUPTOOLS_SCM_PRETEND_VERSION = version;
+
+  buildInputs = [
```
Comment (mweinelt) [inline pkgs/development/python-modules/galois/default.nix:29]:
```suggestion
  nativeBuildInputs = [
```

buildInputs are runtime dependencies, nativeBuildInputs are build time dependencies.

Comment (ChrisPattison) [reply pkgs/development/python-modules/galois/default.nix]: Thanks for the catch. I've fixed this

File: pkgs/development/python-modules/galois/default.nix [hunk 2]
```diff
@@ -1,5 +1,6 @@
 { lib
 , buildPythonPackage
+, setuptools_scm
```
Comment (risicle) [inline pkgs/development/python-modules/galois/default.nix:3]: I think this has been renamed to `setuptools-scm`?

```
error: Function called without required argument "setuptools_scm" at ...
```

Comment (ChrisPattison) [reply pkgs/development/python-modules/galois/default.nix]: Both `setuptools-scm` and `setuptools_scm` work. Is this something that will be deprecated in the future? I renamed it here

Comment (mweinelt) [reply pkgs/development/python-modules/galois/default.nix]: Yes, aliases are the way to deprecate attributes. There is an `allowAliases` switch in `nixpkgs.config` that one can use to disallow aliases and notice these breakages earlier.

Comment (ChrisPattison) [reply pkgs/development/python-modules/galois/default.nix]: Awesome, I will use that in the future

Comment (risicle) [reply pkgs/development/python-modules/galois/default.nix]: :thinking: are you testing your bare branch rather than the merged result with master?

**Comment by risicle** [comment]
@ofborg build python39Packages.galois python310Packages.galois

File: pkgs/development/python-modules/galois/default.nix (hunk 2 above)
Comment (ChrisPattison) [reply pkgs/development/python-modules/galois/default.nix]: Turns out, I was testing on a ~3 week old branch instead of the latest

Comment (ChrisPattison) [reply pkgs/development/python-modules/galois/default.nix]: I would have thought ofborg would have caught that error? Definitely a lesson learned. Sorry about that

**Pull request merged**